packages/backend/src/index.ts

```diff
@@ -347,10 +355,26 @@ export function initIocContainer(
       peerService: await deps.use('peerService'),
       ratesService: await deps.use('ratesService'),
       streamServer: await deps.use('streamServer'),
-      ilpAddress: config.ilpAddress
+      ilpAddress: config.ilpAddress,
+      telemetry: config.enableTelemetry
+        ? await deps.use('telemetry')
+        : undefined
```

Comment:
This was indeed in the accounting service, using 'onDebit', but it moved to connector so that we can collect at packet level. 
I see your scalability concern. Is there another way to collect at packet level?

Reply:
Having another look into this.
We need to:

- collect at packet level,
-  keep both the count and the amount metrics on the same side of a transaction, to avoid discrepancies.

By the current implementation, they are both on the Sending side. I belive the `onDebit` method in the accounting service is on the receiving side of a transaction.

I will explore moving both back to the receiving side. I remember having some trouble with that implementation though, in the local playground both happy-life and cloud-nine fired the metric instead of just one, and the amount ended up doubling if i recall correctly. Does it make sense for that to happen?

Will give it another shot

Reply:
I believe we can use the accounting service and still achieve gathering metrics on the sending side. I'm discussing this with Beni in Slack.

Reply:
In `createAccountToAccountTransfer` in the `packages/backend/src/accounting/service.ts` file there is a `post` method and inside that there we can add a `if (destinationAccount.onDebit)` conditional. If we set our metric to only be collected on when the destination account is being debited then then I believe we will be collecting the accounting, packet level, sending side only, meeting all our objectives.
Like so:
```
post: async (): Promise<void | TransferError> => {
      const error = await postTransfers(pendingTransferIdsOrError)

      if (error) {
        return error
      }

      await Promise.all([
        onDebit(sourceAccount),
        onDebit(destinationAccount.asset)
      ])

      if (destinationAccount.onCredit) {
        const totalReceived = await getAccountReceived(destinationAccount.id)

        if (totalReceived === undefined) {
          throw new Error()
        }

        await destinationAccount.onCredit({
          totalReceived,
          withdrawalThrottleDelay
        })
      }
      
      if (destinationAccount.onDebit) {
        // collect metrics here on destinationAmount
      }
    },
```

Reply:
@JoblersTune is correct here. Some context: this works because in cases where we are doing sending over Interledger, the destination account will be a `Peer`, that _does_ have an `onDebit` method. See [here](https://rafiki.dev/concepts/accounting/accounts-and-transfers/#interledger).

If you want to be more explicit, you could try using `if (destinationAccount instanceof Peer)` to get the same effect

Reply:
I already started moving the codebase back to the accounting service, and using the suggested conditionals to filter sending transactions.

@mkurapov  just to clarify, so if the payment is done within the same rafiki instance, there would be no onDebit method on the destinationAccount? 

In a broader view, I do not think local payments are necessarily the best indicator of ILP network growth. 
It could however be a factor in having a feel about the activity within the network. How many users there are, or at least how active the served users are from a transaction standpoint. That would be the argument I see in supporting those local payments.

That being said, not having that condition means that we need another way to differentiate between sending and receiving transactions, so we react just on one side (suggesting sending side).  

If we don't, we could have another approach:
Having an additional buffer for metrics before they are sent, that is also able to de-duplicate amount metrics that come up both from the sending and from the receiving side of a transaction.
This is however might have diminishing returns in terms of complexity vs achievement.

Thanks for your insights

Reply:
update: latest commit moved everything back to the accounting service.
Please have a look

Reply:
@beniaminmunteanu 
> @mkurapov just to clarify, so if the payment is done within the same rafiki instance, there would be no onDebit method on the destinationAccount and the telemetry instrumentation code would not execute?

That's right, since the source/debit account is the outgoing payment and the destination/credit account is the incoming payment that only has the onCredit method, and not the onDebit (since it doesn't make too much sense to ever debit an incoming payment

Reply:
@beniaminmunteanu 

> I do not think local payments are necessarily the best indicator of ILP network growth.
I agree with you, if we wanted to track that, I think that should belong in its own metric.